### PR TITLE
RD-2000 Enable `consistent-type-imports` eslint rule

### PR DESCRIPTION
**Pull request by kubama**
## Description
Enabled `consistent-type-imports` eslint rule as an alternative to enabling `importsNotUsedAsValues` TS compiler setting. The benefit of this solution is that, as opposed to the original solution outlined in RD-2000, this eslint rule is auto-fixable, while it fails the build in the same cases.

## Screenshots / Videos
N/A

## Checklist

- [X] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I verified that all tests and checks have passed.
- [X] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [X] I added proper labels to this PR.

## Tests
N/A

 

## Documentation
N/A

### CODE REVIEW

File: backend/migration.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
-import { Umzug, SequelizeStorage } from 'umzug';
-import { chain, each, map, last } from 'lodash';
-import { DataTypes } from 'sequelize';
+import { chain, each, last, map } from 'lodash';
+import type { QueryInterface, Sequelize, DataTypes } from 'sequelize';
```
Comment (qooban) [inline backend/migration.ts:2]: I'm not sure about `DataTypes` here. 
I suspect (I'm not 100% sure) it was imported with just `import` (and not `import type`) intentionally.

**Review by qooban — COMMENTED** [review]
> Enabled `consistent-type-imports` eslint rule as an alternative to enabling importsNotUsedAsValues TS compiler setting. The benefit of this solution is that, as opposed to the original solution outlined in [RD-2000](https://cloudifysource.atlassian.net/browse/RD-2000), this eslint rule is auto-fixable, while it fails the build in the same cases.

Good idea! Making it auto-fixable is important benefit in this case 👍🏻

File: backend/migration.ts (hunk 1 above)
Comment (kubama) [reply backend/migration.ts:2]: I was surprised here as well. It is used with `typeof` operator (and it's the only usage), so I would expect a need to import it as a value. But this is not the case, looks like in such cases `import type` is sufficient and is actually enforced by the eslint rule. Everything compiles fine so I assume we are good here.

Comment (qooban) [reply backend/migration.ts:2]: OK, so if it compiles fine, then are we sure that:

1. TS checks (`check-types`) in Stage and Composer TS are not failing?
2. we get proper support from IDE with DataTypes? (see video below)

https://user-images.githubusercontent.com/5202105/166656866-c976d5a5-5197-464e-aeea-ace41ba177ec.mp4

**Review by qooban — APPROVED** [review]
Added one more comment (see in the open discussion). 

If all is fine regarding TS support for DataTypes, I'm OK with merging this PR.

File: backend/migration.ts (hunk 1 above)
Comment (kubama) [reply backend/migration.ts:2]: I verified both points - no issues.

Comment (qooban) [reply backend/migration.ts:2]: Thanks! So we are good to go 👍🏻

**Pull request merged**